docs/guides/frontend-redux.md

```diff
@@ -0,0 +1,63 @@
+
+Each module is joined in the `store.js`, which is the high-level place where we configure our Redux store. 
+
+The rules for a duck are that A module...
+
+1. MUST `export default` a function called `reducer()`
+2. MUST `export` its action creators as functions
+3. MUST have action types in the form `npm-module-or-app/reducer/ACTION_TYPE`
+4. MAY export its action types as `UPPER_SNAKE_CASE`, if an external reducer needs to listen for them, or if it is a published reusable library
+
+
+## Your first duck - `auth`
+
+In frontend-scaffold, one of our ducks is responsible for managing user authentication. In [`src/auth/ducks`](https://github.com/Code-4-Community/frontend-scaffold/tree/master/src/auth/ducks), you will find everything that you need to allow a user to authenticate with `backend-scaffold`, and the tools to use that data throughout our app. 
+
+To explain this duck, let's walk through what happens when a user signs in to the application. 
+
+First, they enter their email and password, and click submit. Upon that click, our code dispatches a `login` *thunk*. Do not be alarmed by the new buzzword - a thunk is essentially an action which can have asynchronous side effects, for example, reaching out to `backend-scaffold` via an HTTP request to sign in the user. That `login` thunk, and others related to auth, live in [`src/auth/ducks/thunks.ts`](https://github.com/Code-4-Community/frontend-scaffold/tree/master/src/auth/ducks/thunks.ts). A thunk returns a function which makes the async call, then dispatches another action depending on the result. 
+
+Throughout our application, we'll use a utility called `AsyncRequest`. This is a simple yet powerful tool for building asynchronous data into our Redux store. It's not super complicated, and it's not even a separate library - it's simply a file in [`src/utils/asyncRequest.ts`](https://github.com/Code-4-Community/frontend-scaffold/blob/master/src/utils/asyncRequest.ts), and it's only 246 lines. 
+
+The motivation behind AsyncRequest is that all actions, reducers, and state around asynchronous requests are very similar, with the difference being the resulting data payload or error payload for each.]
+
+So, our auth state has one `AsyncRequest` field, `tokens`. It contains the result of either a login or signup action. We store, in our state, whether the request is `NotStarted`, `Loading`, `Loaded`, or `Failed`. All `AsyncRequest`s have these same properties. And depending on those properties, we can access the resulting data payload or error payload. 
+
+Let's get back to our story. In our login thunk, we call `authClient.login()`. If the result is successful, we then `dispatch(authenticateUser.loaded(response))`. Here, we're using part of `AsyncRequest`, specifically the `generateAsyncActions` utility which we use in `src/auth/ducks/actions.ts`. This utility gives us the `loaded` and `failed` actions we use depending on the result of the async request in our thunk. 
```

Comment:
What does the `loaded()` function do here (I know you explained it above, but it seemed more like an enum rather than a function)? And what is the `authenticateUser` object for?

Reply:
Each Asynchronous Request has the same expected behavior in each of these enumerated cases. 

`authenticateUser` gives us four actions that update the `tokens` part of our state (which itself is of type `AsyncRequest<TokenPayload, any>`. 

So, `authenticateUser.loaded(response)` generates an **action**. This action contains the API response, and is of type `loaded`. That action is caught by our reducer, which appropriately updates our `tokens` field of the store. Because it is a `loaded` action, our reducer will set `tokens: AsyncRequest<TokenPayload, any>` to `Completed`, allowing us to access the resulting data payload anywhere in the app.